shm/Store.cpp

```diff
@@ -1,33 +1,152 @@
 #include "Store.hpp"
 
 #include <algorithm>
+#include <random>
 
-Store::Store(std::shared_ptr<Time>& time)
-    : time_(time) {
-    time_->addObserver(this);
+#include "Alcohol.hpp"
+#include "Fruit.hpp"
+#include "Item.hpp"
+
+Store::Store(std::shared_ptr<Time>& time) {
+    time_ = time;
+    time->addObserver(this);
+    market_ = makeStock(generateFruits(), generateAlcos(), generateItems());
```

Comment:
I would rather move `generateFruits(), generateAlcos(), generateItems()` into the body of `makeStock()` method. It looks strange when called directly as parameters. Results of these three methods are used only within `makeStock()` so I think it's better to call them inside of `makeStock`.